fsw/cfe-core/src/msg/cfe_msg_time.c

```diff
@@ -0,0 +1,134 @@
+/*
+**  GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**  Copyright (c) 2006-2019 United States Government as represented by
+**  the Administrator of the National Aeronautics and Space Administration.
+**  All Rights Reserved.
+**
+**  Licensed under the Apache License, Version 2.0 (the "License");
+**  you may not use this file except in compliance with the License.
+**  You may obtain a copy of the License at
+**
+**    http://www.apache.org/licenses/LICENSE-2.0
+**
+**  Unless required by applicable law or agreed to in writing, software
+**  distributed under the License is distributed on an "AS IS" BASIS,
+**  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**  See the License for the specific language governing permissions and
+**  limitations under the License.
+*/
+
+/******************************************************************************
+ *  Time field access functions
+ */
+#include "cfe_sb.h"
+#include "cfe_msg.h"
+#include "cfe_error.h"
+#include <string.h>
+
+/******************************************************************************
+ * Set message time - See API and header file for details
+ */
+int32 CFE_MSG_SetMsgTime(CFE_SB_MsgPtr_t MsgPtr, CFE_TIME_SysTime_t NewTime)
+{
+
+    int32 Result = CFE_SB_WRONG_MSG_TYPE;
+
+    CFE_SB_TlmHdr_t *TlmHdrPtr;
+
+    /* declare format specific vars */
+    #if (CFE_MISSION_SB_PACKET_TIME_FORMAT == CFE_MISSION_SB_TIME_32_16_SUBS)
+    uint16 LocalSubs16;
+    #elif (CFE_MISSION_SB_PACKET_TIME_FORMAT == CFE_MISSION_SB_TIME_32_32_M_20)
+    uint32 LocalSubs32;
+    #endif
```

Comment:
Yup, these will go away eventually since we'll be able to source select with the updates.  Minimizing changes this round just to get the capabilities in place.